packages/flutter_tools/lib/src/commands/apk.dart

```diff
@@ -64,12 +69,27 @@ class _ApkBuilder {
 
     String buildTools = '$androidSdk/build-tools/$_kBuildToolsVersion';
     _aapt = new File('$buildTools/aapt');
+    _dx = new File('$buildTools/dx');
     _zipalign = new File('$buildTools/zipalign');
     _jarsigner = 'jarsigner';
   }
 
   bool checkSdkPath() {
-    return (_androidJar.existsSync() && _aapt.existsSync() && _zipalign.existsSync());
+    return (_androidJar.existsSync() && _aapt.existsSync() && _dx.existsSync() && _zipalign.existsSync());
+  }
+
+  void compileClassesDex(File classesDex, List<File> jars) {
+    // Note: I don't know what these args do, and dx has no docs. I'm just
+    // copying chromium's build system.
+    List<String> packageArgs = [_dx.path,
+      '--dex',
+      '--force-jumbo',
```

Comment:
I found http://tools.android.com/download/adt-21-preview which suggests it just allows more strings in the binary.